frontend/src/hooks/profile/useFetchProfileDetails.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState, useCallback } from "react";
+import { UserProfile } from "../../../utils/types";
+import axios from "axios";
+import encodeCredentials from "../../../utils/credentialsencoder";
+
+const useFetchProfileDetails = (id: number) => {
+	const [userProfileData, setUserProfileData] = useState<UserProfile | null>(
+		null
+	);
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [error, setError] = useState<string | null>(null);
+	const BASE_URL = import.meta.env.VITE_API_BASE_URL;
+	const username = "chitchat-test";
+	const password = "7lBeZIE=";
+	const credentials = encodeCredentials(username, password);
+
```

Comment:
@Presson-coder Can we fetch the profile details without hardcoding the credentials? The code should use the current user's credentials to make API requests instead of a hard-coded credential.

Can we save the credentials in a secure storage, retrieve them and pass them to the hook whenever it is called?

Reply:
Oww on that one I needed the endpoint for login which I assume its not there yet. So when its there ,we will make that data dynamic and we will display the data of the currently logged in user and subsequently make requests using their respective credentials.

So on that one I was just doing it that way so that I make requests with the user you created

Reply:
We have a login endpoint here: http://chitchat-api.vndprojects.com/api-auth/login/

I don't have a problem with testing auth using this user account. However, to test more securely, I suggest passing the credentials of whatever user we're using without hard-coding them.

@Presson-coder is this possible?

Reply:
Ok sure thanks, I didn't know that the endpoint was there. Let me work on  that and update you.